servers/src/test/java/tachyon/worker/block/meta/StorageDirTest.java

```diff
@@ -49,11 +57,101 @@
+    mDir = StorageDir.newStorageDir(mTier, TEST_DIR_INDEX, TEST_DIR_CAPACITY, TEST_DIR_PATH);
     mBlockMeta = new BlockMeta(TEST_BLOCK_ID, TEST_BLOCK_SIZE, mDir);
     mTempBlockMeta = new TempBlockMeta(TEST_USER_ID, TEST_TEMP_BLOCK_ID, TEST_BLOCK_SIZE, mDir);
   }
 
+  private StorageDir newStorageDir(File testDir) {
+    return StorageDir.newStorageDir(mTier, TEST_DIR_INDEX, TEST_DIR_CAPACITY,
+        testDir.getAbsolutePath());
+  }
+
+  private File newBlockFile(File dir, String name, int lenBytes) throws IOException {
+    File block = new File(dir, name);
+    block.createNewFile();
+    byte[] data = TestUtils.getIncreasingByteArray(lenBytes);
+    TestUtils.writeBufferToFile(block.getAbsolutePath(), data);
+    return block;
+  }
+
+  private File newTestDir() {
+    File testDir = Files.createTempDir();
+    LOG.debug("new TEST_DIR = {}", testDir.getAbsolutePath());
+    return testDir;
+  }
+
+  @Test
+  public void initializeMetaNoExceptionTest() throws IOException {
+    File testDir = newTestDir();
+
+    int nBlock = 10;
+    long availableBytes = TEST_DIR_CAPACITY;
+    for (int blockId = 0; blockId < nBlock; blockId ++) {
+      int blockSizeBytes = blockId + 1;
+      newBlockFile(testDir, String.valueOf(blockId), blockSizeBytes);
+      availableBytes -= blockSizeBytes;
+    }
+
+    mDir = newStorageDir(testDir);
+    Assert.assertEquals(TEST_DIR_CAPACITY, mDir.getCapacityBytes());
+    Assert.assertEquals(availableBytes, mDir.getAvailableBytes());
+    for (int blockId = 0; blockId < nBlock; blockId ++) {
+      Assert.assertTrue(mDir.hasBlockMeta(blockId));
+    }
+
+    FileUtils.deleteDirectory(testDir);
+  }
+
+  private void assertStorageDirEmpty(File dirPath, StorageDir dir, long capacity) {
+    Assert.assertEquals(capacity, dir.getCapacityBytes());
+    Assert.assertEquals(capacity, dir.getAvailableBytes());
+    Assert.assertTrue(dir.getBlockIds().isEmpty());
+    File[] files = dirPath.listFiles();
+    Assert.assertNotNull(files);
+    Assert.assertEquals(0, files.length);
+  }
+
+  @Test
+  public void initializeMetaDeleteInappropriateFileTest() throws IOException {
+    File testDir = newTestDir();
+
+    newBlockFile(testDir, "block", 1);
+
+    mDir = newStorageDir(testDir);
+    assertStorageDirEmpty(testDir, mDir, TEST_DIR_CAPACITY);
+
+    FileUtils.deleteDirectory(testDir);
+  }
+
+  @Test
+  public void initializeMetaDeleteInappropriateDirTest() throws IOException {
+    File testDir = newTestDir();
+
+    File newDir = new File(testDir, "dir");
+    boolean created = newDir.mkdir();
+    Assert.assertTrue(created);
+
+    mDir = newStorageDir(testDir);
+    assertStorageDirEmpty(testDir, mDir, TEST_DIR_CAPACITY);
+
+    FileUtils.deleteDirectory(testDir);
+  }
+
+  @Test
+  public void initializeMetaBlockLargerThanCapacityTest() throws IOException {
+    File testDir = newTestDir();
+
+    long blockId = 1001;
+    int capacity = 10;
+    newBlockFile(testDir, String.valueOf(blockId), capacity + 1);
```

Comment:
NIT: how about create a BlockFile of size TEST_BLOCK_SIZE, and make the StorageDir of capacity TEST_BLOCK_SIZE -1.